cli/cliui/agent.go

```diff
@@ -121,26 +132,101 @@ func Agent(ctx context.Context, writer io.Writer, opts AgentOptions) error {
 		resourceMutex.Unlock()
 		switch agent.Status {
 		case codersdk.WorkspaceAgentConnected:
+			// NOTE(mafredri): Once we have access to the workspace agent's
+			// startup script logs, we can show them here.
+			// https://github.com/coder/coder/issues/2957
+			if agent.DelayLoginUntilReady && !opts.SkipDelayLoginUntilReady {
+				switch agent.LifecycleState {
+				case codersdk.WorkspaceAgentLifecycleCreated, codersdk.WorkspaceAgentLifecycleStarting:
+					select {
+					case <-messageAfterDone:
+						showMessage()
+					default:
+						// This state is normal, we don't want
+						// to show a message prematurely.
+					}
+				case codersdk.WorkspaceAgentLifecycleReady:
+					return nil
+				default:
+					showMessage()
+				}
+				continue
+			}
 			return nil
 		case codersdk.WorkspaceAgentTimeout, codersdk.WorkspaceAgentDisconnected:
 			showMessage()
 		}
 	}
 }
 
-func waitingMessage(agent codersdk.WorkspaceAgent) string {
-	var m string
+type message struct {
+	Spin         string
+	Prompt       string
+	Troubleshoot bool
+	Error        bool
+}
+
+func waitingMessage(agent codersdk.WorkspaceAgent) (m *message) {
+	m = &message{
+		Prompt: "Don't panic, your workspace is booting up!",
```

Comment:
Can we add directions on how to bypass this e.g.
```
To skip this check, use the `--no-wait` argument
```

(Or whatever we end up calling it

Reply:
Not for this message (this state is before agent is connected so we can't connect yet).

We could add it for the other message, but I wonder if it would encourage bypassing the prompt, esp. without understanding what the flag does. You just go `Ctrl+C -> --no-wait` yay I'm in (and not everything is working). Would it suffice to explain this in the troubleshooting link?

I think this need will also be alleviated by streaming the startup log once we have it available.

Reply:
> Would it suffice to explain this in the troubleshooting link?

That's fine either